orchestrator/lambdas/layers/heimdall_utils/heimdall_utils/utils.py

```diff
@@ -96,18 +100,28 @@ def parse_timestamp(timestamp: Optional[str] = None) -> str:
     Notes:
         - The function uses the current system time to calculate the 3-month offset.
         - All returned timestamps are in UTC (denoted by the 'Z' suffix).
-        - The function logs a warning message when generating a default timestamp.
     """
-    format = "%Y-%m-%dT%H:%M:%SZ"
     try:
-        if timestamp and bool(datetime.strptime(timestamp, format)):
+        if timestamp and is_valid_timestamp(timestamp):
             return timestamp
     except (TypeError, ValueError):
         log.error("Timestamp is invalid")
```

Comment:
What errors are these trying to catch? Would it have been from the old `bool(datetime.strptime(timestamp, format))` line?

If the error is still a consideration, would it make more sense to handle this error in `is_valid_timestamp` so it's closer to the line that would throw the error?

Reply:
It makes sense to move it to `is_valid_timestamp`. This was catching invalid timestamp strings. 
For example is_valid_timestamp("2024") would throw a ValueError